modules/jolt_physics/spaces/jolt_space_3d.cpp

```diff
@@ -63,6 +63,11 @@ constexpr double DEFAULT_SOLVER_ITERATIONS = 8;
 } // namespace
 
 void JoltSpace3D::_pre_step(float p_step) {
+	while (needs_optimization_list.first()) {
+		JoltShapedObject3D *object = needs_optimization_list.first()->self();
+		object->commit_shapes(true); // This will also removed the object from the list.
```

Comment:
commit_shapes has 2 early outs that don't remove the body from the queue in which case this becomes an infinite loop. Is this a problem (I can't see if these early outs can be taken)?

Reply:
Three early outs, if you include the error, neither of which should rightfully ever happen. But yes, it's probably better if I just redundantly remove the object in the loop as well, just to be safe.